test/python/qasm3/test_export.py

```diff
@@ -148,16 +146,24 @@ def test_registers_as_aliases(self):
             [
                 "OPENQASM 3;",
                 'include "stdgates.inc";',
-                "qubit[10] _all_qubits;",
-                "let first_four = _all_qubits[0:3];",
-                "let last_five = _all_qubits[5:9];",
-                # The exporter does not attempt to output steps.
-                "let alternate = _all_qubits[0:0] ++ _all_qubits[2:2] ++ _all_qubits[4:4] ++ _all_qubits[6:6] ++ _all_qubits[8:8];",
-                "let sporadic = _all_qubits[4:4] ++ _all_qubits[2:2] ++ _all_qubits[9:9];",
+                "qubit _qubit0;",
+                "qubit _qubit1;",
+                "qubit _qubit2;",
+                "qubit _qubit3;",
+                "qubit _qubit4;",
+                "qubit _qubit5;",
+                "qubit _qubit6;",
+                "qubit _qubit7;",
+                "qubit _qubit8;",
+                "qubit _qubit9;",
+                "let first_four = {_qubit0, _qubit1, _qubit2, _qubit3};",
+                "let last_five = {_qubit5, _qubit6, _qubit7, _qubit8, _qubit9};",
+                "let alternate = {first_four[0], first_four[2], _qubit4, last_five[1], last_five[3]};",
+                "let sporadic = {alternate[2], alternate[1], last_five[4]};",
```

Comment:
It looks like a bit is referred to by its most recently created alias. Is this an implementation detail? Is it just an artifact of the implementation?

Reply:
Mostly an implementation detail - I (vaguely) commented on the behaviour here: https://github.com/Qiskit/qiskit-terra/blob/6160ef6ea49f243052a3e5445706e0f8e2dfb6f8/qiskit/qasm3/exporter.py#L769-L775